### DISCUSSION TITLE
Better default settings

### DISCUSSION THREAD

**Discussion by 2TallTyler** [OpenTTD/OpenTTD#8393] [General]
This Discussion has come to an end with the implementation of #8463 . Thank you all for participating!

----

I've been thinking recently of ways to improve the OpenTTD experience for new players who have not yet waded into the settings menu we have all tweaked to our liking. This means changing default settings. I understand that this can be a contentious topic, but I think it’s worth addressing, because first impressions make an impact.

This builds on my [recent PR](https://github.com/OpenTTD/OpenTTD/pull/8352) which enables autorenew by default.

My suggestions are based not just on my favorite settings, but on what I changed when I introduced two new players to the game this fall (neither of whom normally play strategy or tycoon games, but who were immediately hooked).

I am making a separate comment for each change, so please comment in replies to that comment. If you have suggestions for other settings, I suppose this is as good a place as any to debate those.

EDIT: The agreed-upon changes are in #8463.

**Comment by 2TallTyler** [comment]
**Group expenses in company finance window 
(default off, suggest on)**

The default ungrouped expenses are hard to read and virtually useless for analyzing a company’s performance. Grouped expenses are much, much better.

I’m not aware of any use cases for ungrouped expenses, so we might as well discuss having grouped expenses be the only option, allowing us to remove the setting entirely.

Ungrouped (I think the "Other" expenses are local authority bribes):
![group_expenses_off](https://user-images.githubusercontent.com/55058389/102420031-1d763b80-3fcf-11eb-867c-c1ed233a0f49.PNG)

Grouped:
![group_expenses_on](https://user-images.githubusercontent.com/55058389/102420033-1ea76880-3fcf-11eb-922a-f57cac0c172e.PNG)

**Reply by JGRennison** [reply]
Should this really be a setting at all?
Why not just hard-code it to "on"?

**Reply by ldpl** [reply]
@JGRennison Well, idk about defaults but I like that there is an option. IMO grouped just clutters the window with useless info. At the very least it should've grouped incomes with costs for each transport type.

**Reply by kabal2020** [reply]
> @JGRennison IMO grouped just clutters the window with useless info. At the very least it should've grouped incomes with costs for each transport type.

Like the idea of this being on (I didn't realise this was a setting), maybe improve the descriptions on the 'Total' lines, 'income total', 'costs total',  'misc costs total' etc.. 

This might clutter it more though?

**Reply by embeddedt** [reply]
As someone who has played for a few years I find grouped to have far too much noise. I much prefer the ungrouped setting.

I wonder if having a dialog asking the user which mode they prefer is better. (This goes for all contentious settings).

**Reply by kabal2020** [reply]
Out of interest, wonder if many people actually pay much attention to this screen, does everyone just click it away when it comes up (or turn it off from popping up using the setting?)

**Reply by 2TallTyler** [reply]
I think setup dialogs would be overwhelming to new players who don't understand the game to know which option they prefer. It's easy enough to go into the settings to change it.

**Comment by 2TallTyler** [comment]
**Quick creation of vehicle orders 
(default off, suggest on)**

By default, players need to click Go To before clicking on each station in an order list. Turning this on makes Go To already pressed when a new vehicle’s Orders window is first opened (but  not if opened later once the vehicle has orders), AND keeps it active between assigning stations to the order list. This eliminates a lot of clicking and frustration.

**Comment by 2TallTyler** [comment]
**New orders are non-stop by default 
(default off, suggest on)**

There are use cases for implicit orders, but new players generally intend vehicles to go to the next stop, without making any other stops, passing Go, or collecting $200. Non-stop orders do exactly that.

**Reply by JGRennison** [reply]
Implicit orders interact quite badly with cargodist, to the point where it can create performance problems.

I personally would be happy to see implicit orders removed entirely. (See also: https://github.com/JGRennison/OpenTTD-patches/commit/4c8b358ffe48fe26b709d320b1b6126e13c681fc)

**Reply by michicc** [reply]
Removing implicit orders doesn't make sense to me, unless you actually meant removing stopping orders.

**Comment by 2TallTyler** [comment]
**Towns allowed to build level crossings
(default on, suggest off)** 

Since 1.9, towns have been able to build bridges over rails. Given the tendency of level crossings to destroy road vehicles, I suggest turning these off and letting towns rely on bridges to grow.

Note that town-built bridges over tracks are limited to 4 tiles in length (not including ramp tiles, so in total the bridge 6 tiles). This would block growth across a wide area of tracks, but a level crossing more than 4 tracks wide would be highly undesirable anyway, so I don’t consider this a loss.

**Reply by JGRennison** [reply]
> Given the tendency of level crossings to destroy road vehicles

Slightly off-topic, but I think that this is the more fundamental problem which ought to be addressed.
Preventing towns from creating level crossings is treating a secondary symptom and not the root of the problem.

Bridges require more space and have more placement constraints than level crossings, so this would likely reduce the ability for towns to grow past rail tracks.

**Reply by Andrew350** [reply]
IMO it's a part of the learning process that new players experience road vehicles being destroyed on level crossings; it's a hard lesson on why grade separation is better. Allowing towns to build level crossings by default increases the chance of a new player reusing that infrastructure instead of building their own, thus more chance to learn early on what can happen when train meets bus.

Yes, most of us have it turned off, but that's (probably) because we learned the hard way why level crossings are bad and want to avoid them at all costs.

**Reply by 2TallTyler** [reply]
@Andrew350, I'm not sure I follow your reasoning. If a feature is frustrating and shouldn't be used, why would we want to encourage new players to discover it early in their experience with the game? Surely it would be better to follow JGR's suggestion and fix the source of the frustration, or at least avoid players discovering it in their first hour of gameplay. For all we know, they may not even be aware that they can build road bridges. Not really the welcoming experience we should be aiming for, IMO.

@JGRennison, I'm mostly a trunk player but I know your patchpack includes some level crossing fixes. Do any take an approach that would be suitable for trunk or would be a start-from-scratch process?

**Reply by Moth-Tolias** [reply]
@Andrew350 i'm not familiar with grade separation - what is that?

**Reply by JGRennison** [reply]
@Moth-Tolias It's a general term for crossings on separate levels such as bridges, tunnels, etc.

@2TallTyler Closing adjacent crossings is quite mild and inoffensive, but only has a small impact on they problem. The safer level crossings mechanism would generate too much bikeshedding in the absence of a consensus for the need for improvement, and I would be surprised if it or similar got into trunk.

**Reply by Andrew350** [reply]
> Surely it would be better to follow JGR's suggestion and fix the source of the frustration

@2TallTyler If it really is that frustrating, then I fully agree. I guess my point is that I don't see this as much of an issue, more a matter of convenience for experienced players. And if there really is a problem with level crossings, then it should be dealt with more directly instead of simply disabling them. What that solution might be for now I don't know. In an ideal world we would have in-game mini tutorials to teach players about grade separation or placing signals in a way to make level crossings safer (among many, many other things), but that's all way out of scope for discussing default settings.

**Comment by 2TallTyler** [comment]
**Inflation 
(default on, suggest off)**

My understanding of inflation is that it’s designed to counteract the rapid technological progress of the base game, which has a default time span from 1920 to 2090. However, when used with NewGRFs which don’t advance in speed and capacity at the same rate (especially prior to 1920), it can quickly make games unprofitable.

With the understanding that experienced players can change their own settings, I suggest that turning it off by default would be ideal for new players for two reasons:
- With default settings, money in OpenTTD becomes meaningless so quickly that a new player is unlikely to miss inflation.
- New players are likely to start using NewGRFs before they fully understand the game’s economy or advanced settings, and won’t know that they need to disable Inflation for many NewGRF games.

A [2019 scan of all advertised servers](https://citymania.org/tools/serverstat) showed that 60% of the 183 servers successfully queried had inflation turned off.

**Reply by TrueBrain** [reply]
I think there is a larger problem at play here, that inflation no longer makes sense (what you also point out, and which we can see in other tickets too, like "groundhog year" or PRs like #7589). So changing it to "off" basically deprecates the game mechanic, which might be just what we need to redesign it completely.

**Reply by 2TallTyler** [reply]
@TrueBrain I agree with your reasoning. In lieu of someone willing to redesign it at the moment, is turning it off by default an acceptable interim measure or just a way to shove the problem under the rug?

**Comment by 2TallTyler** [comment]
**Breakdowns
(default Reduced, suggest None)**

Breakdowns are another feature that [CityMania's server survey](https://citymania.org/tools/serverstat) showed is unpopular among players. In 186 servers, 56.3% of players had breakdowns set to None.

I have not played enough with breakdowns on to have a thorough understanding how they impact beginners. I just turn them off for myself and suggest it to others. I welcome more discussion on this topic.

**Reply by frosch123** [reply]
I consider breakdowns a single-player thing:
On multiplayer servers companies tend to run unattended, when some players are active while others are inactive. This conflicts with vehicle aging and reliability, so multiplayer games that are not of the type "everyone plays at the same time" tend to start with all vehicles introduced and no vehicle aging (read: no reliability, no breakdowns).

I think it would be sad to disable breakdowns by default. It renders a whole lot of game mechanics useless: "max reliability" of vehicles has no meaning, "servicing" has no meaning, "aging" has no meaning, "renewing/replacing" has no meaning, ...

**Reply by JGRennison** [reply]
I think that breakdowns would be more palatable if they were less frequent and/or more interesting.
Even with breakdowns at "reduced" they are unnecessarily irritating.

I think the fact that most players turn it off should be seen as a signal that the breakdown mechanics as currently implemented are in need of improvement, not necessarily that breakdowns should be off by default.

(There are various "improved breakdowns" patches floating around which make running with breakdowns enabled more interesting and playable. One of these with various changes is in my branch.)

**Reply by James103** [reply]
Could there be a configurable chance for vehicles to actually break down when they're supposed to? This would enable people to make vehicles actually break down less often than they normally would.

**Reply by ldpl** [reply]
While I agree that it's a single-player thing I also think that they way it's currently implemented is too bad to be the default. Players expect breakdowns to give them a challenge. I.e. having to design clever servicing bays and overtakes, consider reliability and service interval. But and the end of this road is only a disappointment (infinite servint, forced depots, lower train density and lost trains).

P.S. In fact, new players even sometimes think pathfinder is broken because trains keep getting lost without realising that it's just the servicing screwing with them. So in a way it even keeps them from building more coplex and interesting networks.

**Comment by frosch123** [comment]
**Localisation > Savegame date format
(default long (31st Dec 2008), suggest ISO (2008-12-31))**

This sorts savegames in a meaningful way.

**Comment by frosch123** [comment]
**Interface > Viewports > Smooth viewport scrolling
(default Off, suggest On)**

Seeing the map move allows you to guess distances and positions of things relative to each other.

**Reply by 2TallTyler** [reply]
@ldpl Care to elaborate on your downvote?

**Reply by ldpl** [reply]
@2TallTyler Well, idk about defaults but I'm definitely not enabling this myself. Don't need some weird laggy scrolling instead of a convenient instant action. And I usually don't need to judge distance/position because when I scoll I either know them or don't care and even in the rare I need to using minimap is still more convenient. Though I guess it's a matter of a usecase.

**Reply by ldpl** [reply]
Basically, I don't think this feature works well enough te be enabled by default.

**Reply by 2TallTyler** [reply]
I haven't used this feature before, but having now tried it my biggest problem is the slightly-laggy arrow key movement. So I am in agreement that it's perhaps not smooth enough to be on by default.

**Comment by frosch123** [comment]
**Interface > Viewports > Show path reservations for tracks
(default Off, suggest On)**

Understanding signalling is the main issue for new players. The usual recommendation is to switch this on.

**Reply by LordAro** [reply]
Not sure about this one. To me it's a "debug" feature that noticeably changes the look of rail networks that isn't necessary for normal usage.

I think perhaps this needs a different method to viewing the path reservations when required, rather than showing them for all trains all the time. Something similar to Factorio, perhaps :D

**Reply by TrueBrain** [reply]
It is a bit odd if it is considered a "debug" feature if it is in the settings, and more over, if it is a recommendation to new players :)

And your suggestion sounds awesome; but should that prevent us from changing the default for now? I mean, having it on is absolutely better for new players than having it off, not? :)

**Comment by frosch123** [comment]
**Environment > Industries > Allow multiple similar industries per town
(default Off, suggest On)**

Clusters of identical industries create the interesting transport routes.

**Comment by Andrew350** [comment]
**Interface > Construction > Link landscape toolbar to rail/road/water/airport toolbars
(default Off, suggest On)**

Less clicks to access landscaping tools when building routes. Seems useful and unobtrusive to me, but may be a matter of preference.

**Reply by ldpl** [reply]
I may change my mind in the future but I just enabled this thing myself and I think I like it more this way. Even though I rarely use those buttons at all (hotkeys ftw).

**Comment by Andrew350** [comment]
**Disasters/Accidents > Disasters
(default Off, suggest On)**

Maybe I'm just being nostalgic, but I was a bit surprised to see this off by default. I guess it was a casualty of merging the old difficulty settings and it was decided they shouldn't be on for new players (easy setting, I suppose?), but it is a feature I remember having to contend with when I started playing, and it seems like a waste to have it off by default.

I am aware this may not be a popular opinion :P

**Reply by James103** [reply]
I've always played with disasters off, for several reasons:
* The time periods that disasters occur in do not coincide with the time periods that I play in most of the time (my play style is year 2200 or later in order to have all vehicles available)
* Disasters tend to destroy networks over time, reducing profit and requiring regular maintenance. This means that disasters may not be the right option for servers that are running 24/7.
* I prefer disasters occuring in my network to be from my own network design flaws, instead of just randomly occurring from "act of God" causes.

Furthermore, a [2020 scan of all advertised servers](https://citymania.org/tools/serverstat) showed that 92% of the 179 servers successfully queried had disasters turned off.

**Reply by frosch123** [reply]
While I also do not consider enabling disasters a good idea, I disagree with James103's reasoning:
OpenTTD is not a multiplayer-only game (compare >50k downloads per month with 500 people playing online).
* Servers have to deal with unattended play, companies are expected to keep running while noone is actively playing in them.
* Singleplayer does not have this restriction. I even claim that the majority of PC games requires continuous attention.
* Starting in 2200 is common in multiplayer, but discards major parts of the singleplayer game mechanics, like vehicle progression.
* Naturally 92% of servers disable this, but since multiplayer is very different from singleplayer, the statistics in meaningless.

**Reply by Moth-Tolias** [reply]
i always turn off disasters because of how irritating it is to constantly undo the damage done. if rather than destroying infrastructure and vehicles outright, they were instead merely temporarily incapacitated [eg. road vehicles were either eventually returned after their close encounter or automatically replaced, and X-COM didn't destroy my rail along with the ufo] i would probably turn this feature back on.
